esp/esp/program/modules/base.py

```diff
@@ -510,16 +513,23 @@ def _checkGrade(moduleObj, request, tl, *args, **kwargs):
 # Just broke out this function to allow combined deadlines (see meets_any_deadline,
 # meets_all_deadlines functions below).  -Michael P, 6/23/2009
 def _checkDeadline_helper(method, extension, moduleObj, request, tl, *args, **kwargs):
-    if tl != 'learn' and tl != 'teach':
+    if tl != 'learn' and tl != 'teach' and tl != 'volunteer':
         return (True, None)
+    if "require_auth" in kwargs:
```

Comment:
You can just do `..., *args, require_auth=True, **kwargs` in the function definition, and avoid the manual defaulting here.  (But see below, I think you may want to remove it entirely.)